### DISCUSSION TITLE
Can config be changed on-the-fly?

### DISCUSSION THREAD

**Discussion by MunifTanjim** [nvim-neo-tree/neo-tree.nvim#429] [Internal Development]
If I want to test the UI to see how it is rendered for different configs, what are my options?

Can config be changed inside each `it(...)` block?

Or do I need run `nvim --headless --noplugin -u tests/....lua -c "lua require('plenary.test_harness')...` command for each config?

**Comment by cseickel** [comment]
Config can be changed on the fly by calling the setup() function again. The caveat here being that I don't think this is 100% clean. In particular, an existing tree may retain state that is related to the prior config.

It should work as desired if you call setup with your new config, then create a new tab and close the original tab. Closing the tab will dispose of all state in that tab.

**Reply by cseickel** [reply]
Actually, you could also directly called `require("neo-tree.sources.manager").dispose_tab(1)` to dispose all data in the first tab. There is also a corresponding `dispose_window` which takes a window id for `position = "current"` trees.

**Reply by MunifTanjim** [reply]
What does the `_clear_state` do? https://github.com/nvim-neo-tree/neo-tree.nvim/blob/2f850cac63888cfed1124f81b1c42b836a4a15f9/lua/neo-tree/sources/manager.lua#L64

**Reply by cseickel** [reply]
Exactly what you want it to do! I forgot that was there. It does clear everything.

I would call that first and then call setup() with your new config.
